src/pages/IdeLoader/index.tsx

```diff
@@ -0,0 +1,225 @@
+        return;
+      }
+      this.alert = { variant, title };
+      this.setState({ alertVisible: true });
+      if (showAlertTimer) {
+        clearTimeout(showAlertTimer);
+      }
+      showAlertTimer = setTimeout(() => {
+        this.setState({ alertVisible: false });
+      }, variant === AlertVariant.success ? 2000 : 10000);
+    };
+    this.hideAlert = (): void => this.setState({ alertVisible: false });
+    // Prepare showAlert as a callback
+    if (this.props.callbacks && !this.props.callbacks.showAlert) {
+      this.props.callbacks.showAlert = (variant: AlertVariant.success | AlertVariant.danger, title: string) => {
+        this.showAlert(variant, title);
+      };
+    }
+  }
+
+  public componentDidUpdate(): void {
+    const { currentStep, hasError } = this.props;
+
+    const current = this.wizardRef.current;
+    if (current && current.state && current.state.currentStep !== currentStep) {
+      current.state.currentStep = currentStep;
+    }
+
+    if (!hasError && this.state.currentRequestError) {
+      this.setState({ currentRequestError: '' });
+    }
+  }
+
+  private getIcon(id: number): React.ReactNode {
+    const { currentStep, hasError } = this.props;
+    if (currentStep > id) {
+      return (<React.Fragment>
+        <CheckCircleIcon className="wizard-icon" color="green" />
+      </React.Fragment>);
+    } else if (currentStep === id) {
+      if (hasError) {
+        return <ExclamationCircleIcon className="wizard-icon" color="red" />;
+      }
+      return (<React.Fragment>
+        <InProgressIcon className={`${styles.rotate} wizard-icon`} color="blue" />
+      </React.Fragment>);
+    }
+    return '';
+  }
+
+  private getSteps(): WizardStep[] {
+    const { workspaceName, currentStep } = this.props;
+    return [
+      {
+        id: 1,
+        name: (<React.Fragment>
+          {this.getIcon(1)}Initializing workspace
+        </React.Fragment>),
+      },
+      {
+        id: 2,
+        name: (<React.Fragment>
+          {this.getIcon(2)}Waiting for workspace to start
+        </React.Fragment>),
+        canJumpTo: currentStep >= 2,
+      },
+      {
+        id: 3, name: (<React.Fragment>
+          {this.getIcon(3)}Looking for runtime machine which contains IDE
+        </React.Fragment>),
+        canJumpTo: currentStep >= 3,
+      },
+      {
+        id: 4,
+        name: (<React.Fragment>
+          {this.getIcon(4)}Open IDE
+        </React.Fragment>),
+        canJumpTo: currentStep === 4,
+      },
+    ];
+  }
+
+  public render(): React.ReactElement {
+    const { workspaceName, workspaceId, ideUrl, hasError } = this.props;
+    const { alertVisible } = this.state;
+
+    if (ideUrl) {
+      const randVal = Math.floor((Math.random() * 1000000) + 1);
+      return (
+        <div style={{ height: '100%' }}>
+          <iframe className='ide-page-frame' src={`${ideUrl}?uid=${randVal}`} />
+        </div>);
+    }
+
+    return (
+      <React.Fragment>
+        {alertVisible && (
```

Comment:
Please have a look at the docs for `AlertGroup` and revise your code. Variable `alertVisible` doesn't make any sense and should be removed along with related code wherever you introduced it in this PR.

Reply:
Let us discuss [this](https://www.patternfly.org/v4/components/alert-group/react). In my case with only one error which I provided here, it does not make sense to do exactly as in the sample for multiple:

```
{this.state.alerts.map(({key, variant, title}) => (
            <Alert
...
          ))}
```
It changes to 
```
{alertVisible && (
...

        )}
```

I do not need an array with only one error.

```<AlertGroup isToast>``` was add for providing the same styles. 

I totally do not agree with you in this case.

@sleshchenko WDYT?

Reply:
@olexii4 personally I prefer to use components in the way documentation says, even for the only message. Furthermore, I agree with @sleshchenko about temporary storage warning, so here we have one more notification.

Reply:
I have done it